### ISSUE TITLE
Argo UI not serving at http://testing-argo.kubeflow.org/

### ISSUE THREAD

**Issue by jlewi** [kubeflow/testing#92]
The kubeflow-testing cluster was recreated yesterday.

/assign jlewi

**Comment by jlewi** [comment]
ingress exists and is using IP address
35.186.244.175

**Comment by jlewi** [comment]
Our DNS record however is using 35.227.214.243

```
gcloud --project=kubeflow-ci compute addresses list
NAME     REGION  ADDRESS        STATUS
argo-ui          35.190.53.122  IN_USE
```

**Comment by jlewi** [comment]
So it looks like our ingress may not be using the reserved ip.

It doesn't look like argo-ui is new or has changed.

```
cloud --project=kubeflow-ci compute addresses describe argo-ui --global
address: 35.190.53.122
creationTimestamp: '2018-02-20T16:07:58.143-08:00'
description: ''
id: '2037685697512539825'
ipVersion: IPV4
kind: compute#address
name: argo-ui
selfLink: https://www.googleapis.com/compute/v1/projects/kubeflow-ci/global/addresses/argo-ui
status: IN_USE
users:
- https://www.googleapis.com/compute/v1/projects/kubeflow-ci/global/forwardingRules/argo-ui
jlewi@jlewi-carbon-glaptop:~/git_kubeflow$ 
```
ingress doesn't appear to be annotated to use static IP

```
apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      ingress.kubernetes.io/backends: '{"k8s-be-30502--c3cb19bff97cde34":"HEALTHY","k8s-be-31094--c3cb19bff97cde34":"HEALTHY"}'
      ingress.kubernetes.io/forwarding-rule: k8s-fw-kubeflow-test-infra-argo-ui--c3cb19bff97cde34
      ingress.kubernetes.io/target-proxy: k8s-tp-kubeflow-test-infra-argo-ui--c3cb19bff97cde34
      ingress.kubernetes.io/url-map: k8s-um-kubeflow-test-infra-argo-ui--c3cb19bff97cde34
    creationTimestamp: 2018-03-29T18:13:48Z
    generation: 1
    name: argo-ui
    namespace: kubeflow-test-infra
    resourceVersion: "39253"
    selfLink: /apis/extensions/v1beta1/namespaces/kubeflow-test-infra/ingresses/argo-ui
    uid: ed4ca860-337c-11e8-a060-42010af001c5
  spec:
    rules:
    - http:
        paths:
        - backend:
            serviceName: argo-ui
            servicePort: 80
          path: /*
  status:
    loadBalancer:
      ingress:
      - ip: 35.186.244.175
kind: List
metadata:
  resourceVersion: "
```

**Comment by jlewi** [comment]
But the annotation should be here.

```
https://github.com/kubeflow/testing/blob/master/test-infra/components/argo.libsonnet#L194
```

**Comment by jlewi** [comment]
So it looks like there are two issues

1. In cloud console the addresses 
   35.190.53.122 & 35.186.244.175 are associated with the same rule
1. DNS record isn't mapping to the correct IP (looks like it never was)

My hunch is the two are related. My guess is when we first deployed it problem 1 occurred; so we got 2 ip addresses associated with the argu-ui and we used the ephmeral one in DNS

Both IPs   35.190.53.122 & 35.186.244.175 are serving the UI.

The fact that the annotation for the GCP static ip is removed might be correlated with the fact that 2 IPs are provisioned.

**Comment by jlewi** [comment]
I updated the DNS records to use http://35.190.53.122

**Comment by jlewi** [comment]
Looks like a bug with the ingress spec. annotations should be inside metadata

```
https://cloud.google.com/kubernetes-engine/docs/tutorials/configuring-domain-name-static-ip#step_2b_using_an_ingress
```

**Comment by jlewi** [comment]
I deleted
   - ingress
   - forwarding rules
   - backend services

**Comment by jlewi** [comment]
Redeployed the ingress

http://testing-argo.kubeflow.org/workflows
is working

Fixes are in #93
